src/algebra/lie/basic.lean

```diff
@@ -371,6 +447,55 @@ def inverse (f : M →ₗ⁅R,L⁆ N) (g : N → M)
               ... = ⁅x, g n⁆ : (h₁ _),
   ..linear_map.inverse f.to_linear_map g h₁ h₂ }
 
+instance : has_add (M →ₗ⁅R,L⁆ N) :=
+{ add := λ f g, { map_lie' := by simp, ..((f : M →ₗ[R] N) + (g : M →ₗ[R] N)) }, }
+
+instance : has_sub (M →ₗ⁅R,L⁆ N) :=
+{ sub := λ f g, { map_lie' := by simp, ..((f : M →ₗ[R] N) - (g : M →ₗ[R] N)) }, }
+
+instance : has_neg (M →ₗ⁅R,L⁆ N) :=
+{ neg := λ f, { map_lie' := by simp, ..(-(f : (M →ₗ[R] N))) }, }
+
+@[norm_cast, simp] lemma coe_add (f g : M →ₗ⁅R,L⁆ N) : ⇑(f + g) = f + g := rfl
+
+lemma add_apply (f g : M →ₗ⁅R,L⁆ N) (m : M) : (f + g) m = f m + g m := rfl
+
+@[norm_cast, simp] lemma coe_sub (f g : M →ₗ⁅R,L⁆ N) : ⇑(f - g) = f - g := rfl
+
+lemma sub_apply (f g : M →ₗ⁅R,L⁆ N) (m : M) : (f - g) m = f m - g m := rfl
+
+@[norm_cast, simp] lemma coe_neg (f : M →ₗ⁅R,L⁆ N) : ⇑(-f) = -f := rfl
+
+lemma neg_apply (f : M →ₗ⁅R,L⁆ N) (m : M) : (-f) m = -(f m) := rfl
+
+instance : add_comm_group (M →ₗ⁅R,L⁆ N) :=
```

Comment:
Can't you extend the `add_comm_monoid` instance on `monoid_hom M N`?
If not, it looks to me like you will have to manually specify the `nsmul` fields to avoid diamonds for Lie modules over `nat`.

Reply:
Understood, now that Sebastien has given us the power I guess I should use it.

Reply:
I don't see how to use `add_monoid_hom.add_comm_monoid` (or `linear_map.add_comm_group` for that matter) though since the instances on the ancestors don't have the right type. Maybe I'm missing a trick?

In any case I believe I've resolved the issue with https://github.com/leanprover-community/mathlib/pull/7225/commits/683906ba2fe193a4b85824c648c9c7789440621b

Reply:
@jcommelin I realised this morning that because `lie_algebra` and `lie_module` both require the scalars to be ring, we can't talk about modules over `ℕ`.

I guess probably still worth future proofing against this diamond in case some day someone relaxes this assumption (maybe it's worth it if you're studying Leibniz algebras?).